pyiron_atomistics/atomistics/structure/has_structure.py

```diff
@@ -0,0 +1,116 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+from abc import ABC, abstractmethod, abstractproperty
+
+"""
+Mixin for classes that have one or more structures attached to them.
+"""
+
+__author__ = "Marvin Poul"
+__copyright__ = (
+    "Copyright 2021, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0"
+__maintainer__ = "Marvin Poul"
+__email__ = "poul@mpie.de"
+__status__ = "production"
+__date__ = "Apr 22, 2021"
+
+
+class HasStructure(ABC):
+    """
+    Mixin for classes that have one or more structures attached to them.
+
+    Necessary overrides are :abstractmethod:`._get_structure()` and
+    :abstractmethod:`._number_of_structures()`.
+
+    :method:`.get_structure()` checks that iteration_step is valid; implementations of
+    :abstractmethod:`._get_structure()` therefore don't have to check it.
+
+    :method:`.get_number_of_structures()` may return zero, e.g. if there's no structure stored in the object yet or a
+    job will compute this structure, but hasn't been run yet.
+
+    Sub classes that wish to document special behavior of their implementation of :method:`.get_structure` may do so by
+    adding documention to it in the "Methods:" sub section of their class docstring.
+
+    The example below shows how to implement this mixin and how to check whether an object derives from it
+
+    >>> from pyiron_atomistics.atomistics.structure.atoms import Atoms
+    >>> class Foo(HasStructure):
+    ...     '''
+    ...     Methods:
+    ...         .. method:: get_structure
+    ...             returns structure with single Fe atom at (0, 0, 0)
+    ...     '''
+    ...     def _get_structure(self, iteration_step=-1, wrap_atoms=True):
+    ...         return Atoms(symbols=['Fe'], positions=[[0,0,0]])
+    ...     def _number_of_structures(self):
+    ...         return 1
+
+    >>> f = Foo()
+    >>> for s in f.iter_structures():
+    ...     print(s)
+    Fe: [0. 0. 0.]
+    pbc: [False False False]
+    cell: 
+    Cell([0.0, 0.0, 0.0])
+    <BLANKLINE>
+
+    >>> isinstance(f, HasStructure)
+    True
+    """
+
+    def get_structure(self, iteration_step=-1, wrap_atoms=True):
+        """
+        Gets the structure from a given iteration step of the simulation (MD/ionic relaxation). For static calculations
```

Comment:
"(MD/ionic relaxation)" is too restrictive. Consider, e.g. Monte Carlo on chemistry. I would change this to "(e.g. MD, ionic relaxation, Monte Carlo, etc...)" or just remove it.

Reply:
(I lean weakly towards removal)

Reply:
Agree, this also came up [here](https://github.com/pyiron/pyiron_contrib/pull/153).  Maybe it's best to deprecate `iteration_step` and rename it to something more generic together with the docstring.

Reply:
Yeah, I like `frame` personally